### PR TITLE
Add EventListener Status with tkn eventlistener list

### PR DESCRIPTION
**Pull request by pratikjagrut**
# Changes

This PR adds the `AVAILABLE` field to the command `tkn el ls` as requested in issue #1042  

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [x] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if functionality changed/added)
- [ ] Run the code checkers with `make check`
- [x] Regenerate the manpages, docs and go formatting with `make generated`
- [x] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/cli/blob/master/CONTRIBUTING.md)
for more details._

# Release Notes

### CODE REVIEW

File: pkg/cmd/eventlistener/testdata/TestListEventListener-Multiple_EventListener.golden
Comment (danielhelfand) [inline pkg/cmd/eventlistener/testdata/TestListEventListener-Multiple_EventListener.golden:1]: Not that it should happen in this pr, but it might be worth looking into in future if it would be helpful to support displaying the reason like `kubectl` does:

```
NAME                          ADDRESS                                                                           AVAILABLE   REASON
github-listener-interceptor   http://el-github-listener-interceptor.petclinic-pipeline.svc.cluster.local:8080   True        MinimumReplicasAvailable
```

File: pkg/cmd/eventlistener/list.go
Comment (pratikjagrut) [reply pkg/cmd/eventlistener/list.go:121]: In this suggested changes `status` will be of type `string` and the status we're fetching is of type `k8s.io/api/core/v1".ConditionStatus`, this change is bound to throw a type mismatch error, `cannot use el.Status.Conditions[0].Status (variable of type "k8s.io/api/core/v1".ConditionStatus) as string value in assignment`.

File: pkg/cmd/eventlistener/testdata/TestListEventListener-Multiple_EventListener.golden
Comment (pratikjagrut) [reply pkg/cmd/eventlistener/testdata/TestListEventListener-Multiple_EventListener.golden:1]: Do you want me to add this reason in this PR or should raise another one?

**Comment by pratikjagrut** [comment]
> * Sign the CLA

Done.

> * Squash your 4 commits into a single commit

I thought of squashing commits in the last with the detailed commit message after PR is done with the suggested changes.

> * Add an e2e test (let me know if any help is needed for this)

OK

File: pkg/cmd/eventlistener/list.go
Comment (pratikjagrut) [reply pkg/cmd/eventlistener/list.go]: Committed changes, resolving changes.

File: pkg/cmd/eventlistener/list.go
Comment (danielhelfand) [reply pkg/cmd/eventlistener/list.go]: So maybe something like this then:

```
status := v1.ConditionStatus("---")
```

File: pkg/cmd/eventlistener/testdata/TestListEventListener-Multiple_EventListener.golden
Comment (danielhelfand) [reply pkg/cmd/eventlistener/testdata/TestListEventListener-Multiple_EventListener.golden:1]: Let's wait and maybe open an issue to discuss it further.

**Comment by danielhelfand** [comment]
> > * Squash your 4 commits into a single commit
> 
> I thought of squashing commits in the last with the detailed commit message after PR is done with the suggested changes.

Sounds good!

> Where can I add e2e? I see a dir `./test/e2e/` but I don't see any eventlistener e2e there.

This would actually be our first e2e test for EventListeners, so it would make sense to add it in a separate folder under `/test/e2e/`.

**Comment by pratikjagrut** [comment]
> This would actually be our first e2e test for EventListeners, so it would make sense to add it in a separate folder under `/test/e2e/`.

As I'm not much aware of the e2e environment/framework here and I will get some insights by referring to present e2es but can you guide me on how to write e2e for EventListeners and which yamls are supposed to be used for creating EventListeners, and what should be the overall flow of the test?

**Comment by danielhelfand** [comment]
> As I'm not much aware of the e2e environment/framework here and I will get some insights by referring to present e2es but can you guide me on how to write e2e for EventListeners and which yamls are supposed to be used for creating EventListeners, and what should be the overall flow of the test?

Let me actually look a bit further into our e2e cluster set up. I actually don't think we install triggers as part of our cluster set up, which would obviously make it not possible to add an e2e test then. But after that, I can provide some test details on what I had mind.

File: pkg/cmd/eventlistener/list.go
Comment (pratikjagrut) [reply pkg/cmd/eventlistener/list.go]: I've updated the status var declaration but is there any specific reason for this except reducing 2 lines to a one-liner code.

**Comment by pratikjagrut** [comment]
Force push for rebasing.

**Comment by pratikjagrut** [comment]
> Let me actually look a bit further into our e2e cluster set up. I actually don't think we install triggers as part of our cluster set up, which would obviously make it not possible to add an e2e test then. But after that, I can provide some test details on what I had mind.

Hey, @danielhelfand any updates on e2e.

**Comment by danielhelfand** [comment]
@pratikjagrut I have opened #1108 to add the triggers component. It just needs to be added and then the test can be developed. An example EventListener is included that could be used for the e2e test.

**Comment by pratikjagrut** [comment]
@danielhelfand any update on e2e.

**Comment by danielhelfand** [comment]
@pratikjagrut Triggers should now be available on the clusters for e2e tests. As far as the flow of the test, an EventListener should be created via `kubectl` ([example for pipeline tests](https://github.com/tektoncd/cli/blob/029870a0e86221b648fbc8e5b26782021d0cb730/test/e2e/pipeline/pipeline_test.go#L59)). A sample EventListener that can be used is [here](https://github.com/tektoncd/cli/blob/029870a0e86221b648fbc8e5b26782021d0cb730/test/resources/eventlistener.yaml). After the EventListener is created, you will want to run a `tkn el ls` command to list the EventListener created ([example of running `tkn` command](https://github.com/tektoncd/cli/blob/029870a0e86221b648fbc8e5b26782021d0cb730/test/e2e/pipeline/pipeline_test.go#L106)). The test should assert whether or not the EventListener `AVAILABLE` status is `true`.

**Comment by pratikjagrut** [comment]
@danielhelfand I wrote e2e review it once.

File: test/e2e/eventListener/eventListener_test.go
Comment (danielhelfand) [inline test/e2e/eventListener/eventListener_test.go:54]: What this test should assert is whether the list command contains the `AVAILABLE` column and that the `AVAILABLE` value is true since this is how the list command will display the value to end users.

File: test/e2e/eventListener/eventListener_test.go
Comment (danielhelfand) [inline test/e2e/eventListener/eventListener_test.go:47]:
```suggestion
	t.Run("Assert if EventListener AVAILABLE status is true", func(t *testing.T) {
```

Comment (pratikjagrut) [reply test/e2e/eventListener/eventListener_test.go]: Done

File: test/e2e/eventListener/eventListener_test.go
Comment (pratikjagrut) [reply test/e2e/eventListener/eventListener_test.go]: I updated the code.

**Comment by pratikjagrut** [comment]
@danielhelfand I'm not sure how this e2e setup is really working. I was looking at build logs there I saw el AVAILABLE status is false but still test is passing. [Logs](https://00e9e64bac53431b06cbb7123666a44af45a89c0c331cfe96c-apidata.googleusercontent.com/download/storage/v1/b/tekton-prow/o/pr-logs%2Fpull%2Ftektoncd_cli%2F1098%2Fpull-tekton-cli-integration-tests-0_10%2F1295390083964211205%2Fbuild-log.txt?qk=AD5uMEvokseAKVTZRnkO0Goj8xE2-fzsqLeNLeikIW2YxY7fxBsZuOUuBJR1V0ztPLJejGUrEZDcn2RlXQny-FJA0BJl-lPClONRgK7vKY2yq9ovABGbxXz9KyzodyPILEI_NUKIJjMUHf3afiaVh92sNsJU7uc-cgVug2JH1TOmub8m7XlJcT5LfgC1PzlbVMSua5gCrtfXQnJGp4DpmgQcR5cG6KvwHPg6bnr21jEgAshVgVsMSKQcRPJLfo4zvXjvIKQ1n814ct-WVa6-oDovNGUXCuWxQ13ET7QvV_Cj-dLVs2eSjej7GoyVg9mX1uMyhXIff-yoPVDQUi2El6cfwL2iaKB8yFXMRjdwzxXgt13mGSeyF_RP0j9sLoW4bUOeYDQWYLDwzyU6hrmzFH-61YamRXTe_vWuvt8A7nwp8bc18Aoyj2DKLYly9R8uNAa4wMq1ecwYDDM5EKyiNIh49H6BqMjBN8Smpf0KijV1IcWsA7n38jiRw3Uuo1tJQbR00G9eIP38CfOVwDW7ltdpjmpZXdQ8Aof6hZiiNUbZgN1yPd178VjkKfOacf0VxXEkYyuPlr3DttSER8cDcXTpivHWt6a0TKNZY084PIth2iuzWY-GkB48gaAHC6hMp3PFcwz2pVp0ZO3u1eo_EdC0J5ofyI4M7o9gixRK0UwVibF280nermZevTWSNxnCtciX-xivt_fCpT5VC7YgKZ2ev23RyrhxjzHBUi9qGJRrMZaATqJ55sEo1wfY9lpbQSZFgBsAHRkJWuvva93V8R7SeLVYRlws5dbe--AISo-g-g3Hu42kRHsnu3JYkBhdW5yfeI7wOVe1W2tEQZ-QNnt26Dp2iCfVBK0Q6nvi28t6pnxJBM4i7D1w3p9oDf_pNzK9u7FEWHMmmZHwuoxyO9AHlTqW8SxB7ymScJeGbHKiyHCrzeEwqZI&isca=1)
```
Running tkn eventlistener list
NAME       AGE              AVAILABLE
listener   52 seconds ago   False
```
I think the failure is due to the absence of [serviceAccount: tekton-triggers-example-sa](https://github.com/tektoncd/cli/blob/master/test/resources/eventlistener.yaml#L20). I did not notice any code to create service account.

Did I make any mistake in the test?

I'm not able to run e2e on my machine, e2e setup is failing at `wait_until_ready 600 pipelinerun/output-pipeline-run || exit 1` so it's kinda hard for me to debug deeper into it.

**Comment by danielhelfand** [comment]
I see your point about the assertion not really testing this properly. Maybe we can try something like the following:

```go
func TestEventListenerE2E(t *testing.T) {
	t.Parallel()
	c, namespace := framework.Setup(t)
	knativetest.CleanupOnInterrupt(func() { framework.TearDown(t, c, namespace) }, t.Logf)
	defer framework.TearDown(t, c, namespace)

	kubectl := cli.NewKubectl(namespace)
	tkn, err := cli.NewTknRunner(namespace)
	assert.NilError(t, err)

	t.Logf("Creating eventlistener in namespace: %s", namespace)
	kubectl.MustSucceed(t, "create", "-f", helper.GetResourcePath("eventlistener.yaml"))

	t.Run("Assert if EventListener AVAILABLE status is false", func(t *testing.T) {
		res := tkn.Run("eventlistener", "list")
		assert.Assert(t, strings.Contains(res.Stdout, "AVAILABLE") && strings.Contains(res.Stdout, "False"))
	})
}
```

I think the reason it is not becoming available is the readiness/liveness probes are failing for the deployment, so let's just assert that it's false for now. We can try and set up a better EventListener example for testing later on.

File: test/e2e/eventListener/eventListener_test.go
Comment (pratikjagrut) [reply test/e2e/eventListener/eventListener_test.go]: Done.

File: test/builder/builder.go
Comment (pratikjagrut) [reply test/builder/builder.go]: Removed this function as we've changed the assertion method.

File: test/builder/builder.go
Comment (pratikjagrut) [reply test/builder/builder.go]: Removed.

File: test/e2e/eventListener/eventListener_test.go
Comment (danielhelfand) [inline test/e2e/eventListener/eventListener_test.go:31]: The build test is failing due to having this const. It's not used so it should be removed:

```
Linting go files...
test/e2e/eventListener/eventListener_test.go:30:2: `teEventListenerName` is unused (deadcode)
	teEventListenerName = "listener"
```

Comment (pratikjagrut) [reply test/e2e/eventListener/eventListener_test.go:31]: Oh, I forgot to remove it :grimacing:

Comment (pratikjagrut) [reply test/e2e/eventListener/eventListener_test.go]: Done :+1: 
Thank you :pray:

**Comment by danielhelfand** [comment]
@pratikjagrut Thanks for all your work on this. Could you please just squash your commits into a single commit and then I think this should be good.

**Comment by pratikjagrut** [comment]
@danielhelfand I squashed all commits into a single commit.

**Comment by pratikjagrut** [comment]
@danielhelfand I think PR also needs LGTM label, tide check is in pending status with message `Not mergeable. Needs lgtm label`.

**Comment by danielhelfand** [comment]
> @danielhelfand I think PR also needs LGTM label, tide check is in pending status with message `Not mergeable. Needs lgtm label`.

Yes, I'll request reviews from others. A second reviewer should apply that label.

File: pkg/cmd/eventlistener/list.go
Comment (piyush-garg) [inline pkg/cmd/eventlistener/list.go:121]: Is the second check `&& len(el.Status.Conditions[0].Status) > 0` required ?

**Review by piyush-garg — APPROVED** [review]
/lgtm

**Pull request merged**
